### DISCUSSION TITLE
enable genie via wsl.conf

### DISCUSSION THREAD

**Discussion by ta5n** [arkane-systems/genie#280] [Q&A] [answered]
I'm using Windows 11 Pro Version 21H2 (OS build 22000.739).

WSL2 on windows 11 provides "wsl.conf" file. It has a [boot] section with a command key to be executed on wsl vm boot. I couldn't start genie via this line.

I tried the following:

```
# /etc/wsl.conf file content
[boot]
command = exec /usr/bin/genie -s
```

after starting ubuntu-20.04 VM, I've queried genie status

```
$ genie -b
no-bottle

$ genie -r
stopped
```

```
# /etc/wsl.conf file content
[boot]
command = exec /usr/bin/genie -i
# this didn't help either.
```

What do you suggest enabling genie at every start?

Thanks

---
Related Microsoft document [here](https://docs.microsoft.com/en-us/windows/wsl/wsl-config#wslconf) below: 

<html>
<body>
<!DOCTYPE html><h3 cid="n2" mdtype="heading" class="md-end-block md-heading" style="box-sizing: border-box; break-after: avoid-page; break-inside: avoid; orphans: 4; font-size: 1.5em; margin-top: 1rem; margin-bottom: 1rem; position: relative; font-weight: bold; line-height: 1.43; cursor: text; white-space: pre-wrap; color: rgb(51, 51, 51); font-family: &quot;Open Sans&quot;, &quot;Clear Sans&quot;, &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-style: initial; text-decoration-color: initial;"><span md-inline="plain" class="md-plain md-expand" style="box-sizing: border-box;">Boot settings</span></h3><p cid="n3" mdtype="paragraph" class="md-end-block md-p" style="box-sizing: border-box; line-height: inherit; orphans: 4; margin: 0.8em 0px; white-space: pre-wrap; position: relative; color: rgb(51, 51, 51); font-family: &quot;Open Sans&quot;, &quot;Clear Sans&quot;, &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-style: initial; text-decoration-color: initial;"><span md-inline="plain" class="md-plain" style="box-sizing: border-box;">The Boot setting is only available on Windows 11.</span></p><p cid="n4" mdtype="paragraph" class="md-end-block md-p" style="box-sizing: border-box; line-height: inherit; orphans: 4; margin: 0.8em 0px; white-space: pre-wrap; position: relative; color: rgb(51, 51, 51); font-family: &quot;Open Sans&quot;, &quot;Clear Sans&quot;, &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-style: initial; text-decoration-color: initial;"><span md-inline="plain" class="md-plain" style="box-sizing: border-box;">Section label: </span><span md-inline="code" spellcheck="false" class="md-pair-s" style="box-sizing: border-box;"><code style="box-sizing: border-box; font-family: var(--monospace); text-align: left; vertical-align: initial; border: 1px solid rgb(231, 234, 237); background-color: rgb(243, 244, 244); border-radius: 3px; padding: 0px 2px; font-size: 0.9em;">[boot]</code></span></p><figure class="md-table-fig" cid="n5" mdtype="table" style="box-sizing: border-box; margin: 1.2em 0px; overflow-x: auto; max-width: calc(100% + 16px); padding: 0px; cursor: default; color: rgb(51, 51, 51); font-family: &quot;Open Sans&quot;, &quot;Clear Sans&quot;, &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-style: initial; text-decoration-color: initial;">

key | value | default | notes
-- | -- | -- | --
command | string | "" | A string of the command that you would like to run when the WSL instance starts. This command is run as the root user. e.g: service docker start Only available for Windows 11.

</figure>
</body>
</html>

**Comment by cerebrate** [comment, marked as answer]
For some reason, the boot-command setting has always been problematic; I haven't been able to get any version of genie to run successfully using it, across multiple releases.

At the moment, the recommended solution for automatically starting genie along with a WSL distro is this:

https://github.com/arkane-systems/genie/wiki/Automatically-starting-genie-on-every-shell-session

**Reply by ta5n** [reply]
Thank you for your reply,
I guess the shell script you referred is for zsh. Would you add bash equivalent of it?

**Reply by cerebrate** [reply]
I don't use bash myself, so I'd not know what needs tweaking to work (if, indeed, anything). I'll happily post a bash version if someone makes one (or they could add it to the page themselves) though.
